src/cpp/src/visual_language/embedding_model.hpp

```diff
@@ -0,0 +1,42 @@
+// Copyright (C) 2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+
+#pragma once
+
+#include <filesystem>
+#include <vector>
+#include <string>
+
+#include "openvino/core/any.hpp"
+#include "openvino/runtime/core.hpp"
+#include "openvino/runtime/tensor.hpp"
+#include "openvino/runtime/infer_request.hpp"
+#include "openvino/runtime/properties.hpp"
+
+#include "visual_language/vlm_config.hpp"
+
+#include <openvino/openvino.hpp>
+#include "visual_language/processor_config.hpp"
+#include "visual_language/vlm_model_type.hpp"
+
+namespace ov {
+namespace genai {
+
+class EmbeddingsModel {
+public:
+    EmbeddingsModel(const std::filesystem::path& model_dir,
+                    const VLMConfig& vlm_config,
+                    const std::string& device,
+                    const ov::AnyMap& properties);
+
+    ov::Tensor infer(ov::Tensor input_idx);
+
+private:
+    void merge_postprocess(std::shared_ptr<ov::Model> model) const;
+
+    VLMConfig m_vlm_config;
```

Comment:
looks like `VLMConfig` is used only in ctor and we can drop this field

Reply:
removed